### PR TITLE
Added close button for full screen chat

### PR DESCRIPTION
**Pull request by vigilantee**
Task ID:
https://bloomstack.com/desk#Form/Task/TASK-2020-01521

### CODE REVIEW

**Comment by hrwX** [comment]
-    @vigilantee Can we replace the `X` with `fa-times` fontawesome icon ? 
-    It'll look consistent with the other two button too since this close button looks way different than the neighbouring buttons.
         image
-    Since you are anyway adding a button for expanded view, might as well change it for the smaller view

**Comment by Alchez** [comment]
@vigilantee, is this ready for review? Or are you still making changes?

File: frappe/public/js/frappe/chat.js [hunk 1]
```diff
@@ -980,6 +981,8 @@ frappe.chat.message.seen   = (mess, user) => {
 
 frappe.provide('frappe.chat.message.on')
 frappe.chat.message.on.create = function (fn) {
+	console.log("frappe.chat.message.on.create");
+	console.log(fn);
```
Comment (hrwX) [inline frappe/public/js/frappe/chat.js:985]:
```suggestion
```

File: frappe/public/js/frappe/chat.js [hunk 2]
```diff
@@ -1444,9 +1447,21 @@ class extends Component {
 						}
 						else
 							update.typing = frappe._.as_array(update.typing)
+
+						return { ...r, ...update }
+					}
+
+					if(update.seen) {
```
Comment (hrwX) [inline frappe/public/js/frappe/chat.js:1454]:
```suggestion
					if (update.seen) {
```

File: frappe/public/js/frappe/chat.js [hunk 3]
```diff
@@ -1469,9 +1485,18 @@ class extends Component {
 						}
 					} else
 						update.typing = frappe._.as_array(update.typing)
+
+					room  = { ...state.room, ...update }
 				}
 
-				const room  = { ...state.room, ...update }
+				if(update.seen) {
```
Comment (hrwX) [inline frappe/public/js/frappe/chat.js:1492]:
```suggestion
				if (update.seen) {
```

File: frappe/public/js/frappe/chat.js [hunk 4]
```diff
@@ -1550,6 +1579,7 @@ class extends Component {
 			}
 		})
 
+		console.log("binding");
```
Comment (hrwX) [inline frappe/public/js/frappe/chat.js:1582]:
```suggestion
```

File: frappe/public/js/frappe/chat.js [hunk 5]
```diff
@@ -2000,19 +2036,20 @@ class extends Component {
...
 			h("li", null,
 				h("a", { class: props.active ? "active": "", onclick: () => {
-					if (props.last_message) {
-						frappe.chat.message.seen(props.last_message.name);
-					}
-					props.click(props)
+					props.click(props);
+					// if (props.last_message) {
+					// 	frappe.chat.message.seen(props.last_message.name);
+					// }
```
Comment (hrwX) [inline frappe/public/js/frappe/chat.js:2052]:
```suggestion
```

**Review by hrwX — CHANGES_REQUESTED** [review]

**Comment by hrwX** [comment]
Closing this PR since a new PR is up #300

**Pull request closed without merging**